includes/html/forms/rediscover-device.inc.php

```diff
@@ -65,4 +65,4 @@
 ];
 
 header('Content-type: application/json');
-echo json_encode($output, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+echo htmlspecialchars(json_encode($output, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE));
```

Comment:
Can json_encode really be exploited for this?

Reply:
htmlspecialchars() will break json encoding in some situations right?

FYI, you care move the flags `JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE`, they are inlined from an old function and were only used to make the json "look pretty"...

Reply:
Depending on this answer in stackoverflow [here](https://stackoverflow.com/questions/12062146/is-json-encode-sufficient-xss-protection).
json_encode is not enough to protect from XSS. But maybe the suggestion answer can be better than htmlspecialchars.

Reply:
This code is giving an XSS vulnerability in my local computer.
```php
<?php
$output = ["A"=>"<img src='#' onerror=alert(1)>","B"=>"Safe"];
echo json_encode($output);
```

Reply:
Could just add htmlspecialchars around $message on line 64